api/envoy/extensions/filters/http/geoip/v3/maxmind.proto

```diff
@@ -0,0 +1,25 @@
+syntax = "proto3";
```

Comment:
This file should go in a location that matches its type and the path to the implementation. For example, extensions that go in the cache filter live in `envoy/extensions/http/cache` not in `envoy/extensions/filters/http/cache`.

Since this provider is not http specific at all, I would think `envoy/extensions/geoip_providers` would be the place.